tests/rust/chorded/overlapping.rs

```diff
@@ -0,0 +1,459 @@
+use smart_keymap::input;
+use smart_keymap::key;
+use smart_keymap::keymap;
+use smart_keymap::tuples;
+
+use keymap::DistinctReports;
+use keymap::Keymap;
+
+use key::{chorded, composite, keyboard};
+use tuples::Keys4;
+
+type Ctx = composite::Context;
+type Ev = composite::Event;
+type PKS = composite::PendingKeyState;
+type KS = composite::KeyState;
+type CK = composite::ChordedKey<composite::Layered<composite::TapHold<keyboard::Key>>>;
+type AK = composite::ChordedKey<composite::Layered<composite::TapHold<keyboard::Key>>>;
+
+// 4-key keymap
+//   A B C D
+// chords:
+// 0: X X - - => M=0x10
+// 1: X X X - => N=0x11
+// 2: - X X - => O=0x12
+// 3: - - X X => P=0x13
+
```

Comment:
I think it'd be worth adding "X - - X" as a chord, and checking "X - - X" vs "- - X X" as overlapping.

Reply:
Another case to check: a chord with multiple aux keys.

And re-tapping one of the chorded keys. (e.g. "A B", with chord "A B": pressing AB, then releasing A & tapping A (or releasing B & tapping B).